### PR TITLE
Wrap the Rest of the Models

### PR DESCRIPTION
**Pull request by IsaccBarker**
Wrapped the rest of the models. I would have done this in a loop or something instead of having all these separate files, but I wanted to keep it simple before I found out if it would work or not (and not through a duplicate symbol error).

### CODE REVIEW

File: src/epiworldpy/__init__.py [hunk 1]
```diff
@@ -3,3 +3,11 @@
 from ._core import __doc__, __version__, ModelSEIR
 
 __all__ = ["__doc__", "__version__", "ModelSEIR"]
+
+#from ._core import __doc__, __version__, ModelDiffNet, ModelSEIR, ModelSEIRCONN, \
```
Comment (IsaccBarker) [inline src/epiworldpy/__init__.py:7]:
```suggestion
#from ._core import __doc__, __version__, ModelDiffNet, ModelSEIR, ModelSEIRCONN, \
```

**Comment by IsaccBarker** [comment]
The issue is an issue with macro expansion and tokenization. Fixing now and writing tests, should be done soon. After that it's `run_multiple_get_results` for #26.

**Comment by IsaccBarker** [comment]
A things to consider. I'm not doing any data validation *right now* to see if the model is returning the right value (e.g. to check if there was some wires crossed in the binding process and an incorrect model was being run), for the reason below:

*tests/test_model.py*
```python
# TODO: Decide the right way to validate data; how often is the underlying C++
# simulation code changed in a way that will change outputs here? What's the
# tradeoff?
```

There's also the blank test cases for the non-connected models (e.g. `test_seir_simple`, `test_sir_simple`, `test_sird_simple`) because the API to give a model some agents (e.g. `agents_from_adjlist`, `agents_from_edgelist`) haven't undergone binding just yet (see issue #32).

Another thing: I have no idea how I might test `test_diffnet_simple`, because the parameters are a bit opaque. Many of the model parameters aren't documented, and as far as I can see there aren't any examples:

image

**Comment by IsaccBarker** [comment]
@gvegayon Can we consider merging this to allow easier development of #33 and #26?

File: .clang-format
Comment (gvegayon) [inline .clang-format:1]: What is this file for?

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
 README.md: README.qmd
 	quarto render README.qmd
 
-.PHONY: build update
\ No newline at end of file
+.PHONY: compile_commands.json
+compile_commands.json:
+	cmake -S . -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+	cp compile_commands.json ..
+	rm -rf build
```
Comment (gvegayon) [inline Makefile:18]: What does this target do?

File: pyproject.toml [hunk 3]
```diff
@@ -2,27 +2,26 @@
...
-description="Python bindings for epiworld"
+description = "Python bindings for epiworld"
 readme = "README.md"
-authors = [
-  { name = "George G. Vega Yon", email = "g.vegayon@gmail.com" },
-]
 requires-python = ">=3.7"
+dependencies = ["numpy"]
+
+authors = [{ name = "George G. Vega Yon", email = "g.vegayon@gmail.com" }]
```
Comment (gvegayon) [inline pyproject.toml:13]: Probably add yourself here as well?

**Comment by gvegayon** [comment]
So we have the sphinx site included. We should add a GHA to build it. Here is an example of one that also renders quarto files and publishes it using GitHub pages: https://github.com/CDCgov/multisignal-epi-inference/blob/ef872383de148664fad4cbecad96703c2829c5a1/.github/workflows/website.yaml

File: .clang-format
Comment (IsaccBarker) [reply .clang-format:1]: [Automatic formatting of the binding C++ code](https://clang.llvm.org/docs/ClangFormat.html). It keeps things cleaner, and makes things easier to work with in my opinion. It doesn't effect any of the Epiworld code.

File: Makefile (hunk 2 above)
Comment (IsaccBarker) [reply Makefile:18]: It generates a [`compile_commands.json`](https://clang.llvm.org/docs/JSONCompilationDatabase.html) file, which clangd can use to errors in the code to my editor so you don't have to re-compile to check if an error is fixed.

File: pyproject.toml (hunk 3 above)
Comment (IsaccBarker) [reply pyproject.toml:13]: Will do.

**Comment by IsaccBarker** [comment]
Can you ignore memory leak notifications in Codacy for `database.cpp`? It's complaining because we're only `delete`'ing `new`'d memory inside a closure that Codacy can't seem to wrap it's mind around.

**Comment by IsaccBarker** [comment]
> So we have the sphinx site included. We should add a GHA to build it. Here is an example of one that also renders quarto files and publishes it using GitHub pages: https://github.com/CDCgov/multisignal-epi-inference/blob/ef872383de148664fad4cbecad96703c2829c5a1/.github/workflows/website.yaml

Just realized that this should have been it's own commit/PR... Don't know how that got past me, apologies. If you like, I can shuffle things around so that they're separate in the history, but for now I'm assuming we're moving forward like this, so the GHA PR for the Sphinx docs will branch off this.

**Review by gvegayon — APPROVED** [review]

**Pull request merged**